chainimport/headers_import.go

```diff
@@ -0,0 +1,1403 @@
+	return createSpecificImportSource[*ChainImportBlockHeader](
+		options.BlockHeadersSource,
+	)
+}
+
+// createFilterHeadersImportSource creates the appropriate import source for
+// filter headers.
+func (options *ImportOptions) createFilterHeadersImportSource() (HeaderImportSource[*ChainImportFilterHeader], error) {
+	return createSpecificImportSource[*ChainImportFilterHeader](
+		options.FilterHeadersSource,
+	)
+}
+
+// createSpecificImportSource is a private helper function (not a method) that
+// handles the generic implementation.
+//
+// nolint:unparam
+func createSpecificImportSource[T any](source string) (HeaderImportSource[T], error) {
+	// Check if source could be interpreted as a file path
+	// This is our default fallback for any string that doesn't match other
+	// patterns.
+	return &FileHeaderImportSource[T]{
+		Path: source,
+	}, nil
+
+	// For future expansion, additional source types can be added here with
+	// their own pattern detection and source construction.
+}
+
+// ImportOptions defines parameters for the import process.
+type ImportOptions struct {
+	TargetChainParams       chaincfg.Params
+	TargetBlockHeaderStore  headerfs.BlockHeaderStore
+	TargetFilterHeaderStore headerfs.FilterHeaderStore
+
+	// Import source identifiers.
+	BlockHeadersSource  string
+	FilterHeadersSource string
+
+	// Processing parameters.
+	WriteBatchSize int
+	Logger         btclog.Logger
+}
+
+// ImportResult contains statistics about a header import operation.
+type ImportResult struct {
+	// Counts of processed headers.
+	ProcessedCount int // Total number of headers examined
+	AddedCount     int // Number of headers newly added to destination
+	SkippedCount   int // Number of headers already in destination
+
+	// Actual height range processed.
+	StartHeight uint32 // First height processed
+	EndHeight   uint32 // Last height processed
+
+	// Duration information.
+	StartTime time.Time     // When import started
+	EndTime   time.Time     // When import completed
+	Duration  time.Duration // Total time taken
+}
+
+// HeadersPerSecond calculates the processing rate in headers per second as a
+// performance metric. Returns 0 if Duration is zero to avoid division by zero.
+func (r *ImportResult) HeadersPerSecond() float64 {
+	if r.Duration.Seconds() > 0 {
+		return float64(r.ProcessedCount) / r.Duration.Seconds()
+	}
+	return 0
+}
+
+// NewHeadersPercentage calculates the percentage of processed headers that were
+// newly added (not already present in the target). Returns 0 if no headers were
+// processed to avoid division by zero.
+func (r *ImportResult) NewHeadersPercentage() float64 {
+	if r.ProcessedCount > 0 {
+		return float64(r.AddedCount) / float64(r.ProcessedCount) * 100
+	}
+	return 0
+}
+
+// formatDuration converts a duration to a human-readable string format
+// with appropriate units (ns, μs, ms, s, m, h) based on the duration length.
+func formatDuration(d time.Duration) string {
```

Comment:
`time.Duration` has a `String()` method which does this. So can just use `%s` in the log string.